Userland/Libraries/LibVideo/VP9/Parser.cpp

```diff
@@ -933,43 +935,28 @@ DecoderErrorOr<void> Parser::decode_tiles(FrameContext& frame_context)
         }
     }
 
-    auto decode_tile_column = [this, tile_rows](auto& column_workloads) -> DecoderErrorOr<void> {
+    Function<DecoderErrorOr<void>(Vector<TileContext, 1>&)> decode_tile_column = [this, tile_rows](auto& column_workloads) -> DecoderErrorOr<void> {
         VERIFY(column_workloads.size() == tile_rows);
         for (auto tile_row = 0u; tile_row < tile_rows; tile_row++)
             TRY(decode_tile(column_workloads[tile_row]));
         return {};
     };
 
 #ifdef VP9_TILE_THREADING
-    auto const worker_count = tile_cols - 1;
-
-    if (m_worker_threads.size() < worker_count) {
-        m_worker_threads.clear();
-        m_worker_threads.ensure_capacity(worker_count);
-        for (auto i = 0u; i < worker_count; i++)
-            m_worker_threads.append(DECODER_TRY_ALLOC(Threading::WorkerThread<DecoderError>::create("Decoder Worker"sv)));
-    }
-    VERIFY(m_worker_threads.size() >= worker_count);
+    m_process_error = {};
+    m_process_has_error = false;
 
     // Start tile column decoding tasks in thread workers starting from the second column.
-    for (auto tile_col = 1u; tile_col < tile_cols; tile_col++) {
-        auto& column_workload = tile_workloads[tile_col];
-        m_worker_threads[tile_col - 1]->start_task([&decode_tile_column, &column_workload]() -> DecoderErrorOr<void> {
-            return decode_tile_column(column_workload);
-        });
-    }
+    for (auto tile_col = 1u; tile_col < tile_cols; tile_col++)
+        m_thread_pool.submit({ tile_workloads[tile_col], decode_tile_column });
 
     // Decode the first column in this thread.
-    auto result = decode_tile_column(tile_workloads[0]);
+    m_process_error = decode_tile_column(tile_workloads[0]);
```

Comment:
I just realized, doesn't this potentially overwrite the error of one of the threaded workloads if that workload finishes before this column?